z2/src/converter.rs

```diff
@@ -206,13 +208,55 @@ fn get_contract_code(zq1_db: &zq1::Db, address: Address) -> Result<Code> {
         })?
     };
 
-    Ok(Code::Scilla {
-        code: String::from_utf8(code)
-            .map_err(|err| anyhow!("Unable to convert scilla code into string: {err}"))?,
+    let code = String::from_utf8(code)
+        .map_err(|err| anyhow!("Unable to convert scilla code into string: {err}"))?;
+
+    let contract = Contract::from_str(&code)?;
+
+    let scilla_code = Code::Scilla {
+        code: code.clone(),
         init_data: init_data_vec,
         types: BTreeMap::default(),
-        transitions: vec![],
-    })
+        transitions: contract
+            .transitions
+            .iter()
+            .map(|elem| Transition {
+                name: elem.name.clone(),
+                params: elem
+                    .params
+                    .iter()
+                    .map(|params| TransitionParam {
+                        name: params.name.clone(),
+                        ty: format!("{}", params.r#type),
+                    })
+                    .collect(),
+            })
+            .collect(),
+    };
+    // Print all fields of Code::Scilla by matching on scilla_code
+    match &scilla_code {
+        Code::Scilla {
+            code,
+            init_data,
+            types,
+            transitions,
+        } => {
+            println!("Code::Scilla parameters:");
+            println!("Address: {:?}", address);
+            println!("Code: {}", code);
+            println!("Init Data: {:?}", init_data);
+            println!("Types: {:?}", types);
+            println!("Transitions:");
+            for transition in transitions {
+                println!("  Transition Name: {}", transition.name);
+                for param in &transition.params {
+                    println!("    Param Name: {}, Type: {}", param.name, param.ty);
+                }
+            }
+        }
+        _ => {}
+    }
```

Comment:
Is this some debugging that was left behind?

Reply:
I kept it intentionally because I am planning to test persistence conversion with all these PR combined. Currently, it is hard to test and verify if we have more than one branch. The print statement, i will remove later on.